src/qibolab/_core/instruments/emulator/hamiltonians.py

```diff
@@ -122,22 +122,43 @@ def dissipation(self, n: int):
         return self.relaxation(n) + self.dephasing(n)
 
 
+class Coupler(Qubit):
+    """Coupler configuration."""
+
+    coupling: list[float] = Field(default_factory=list)
+    """Coupling between coupler and qubit."""
+
+
 class QubitPair(Config):
     """Hamiltonian parameters for qubit pair."""
 
-    coupling: float
+    coupling: float = 0
     """Qubit-qubit coupling."""
+    coupler: Optional[Coupler] = None
+    """Coupler mediating the interaction."""
 
-    def operator(self, n: int) -> Operator:
-        """Time independent operator."""
+    @staticmethod
+    def _operator(n: int) -> Operator:
         op = tensor_product(
             transmon_destroy(n),
             transmon_create(n),
         ) + tensor_product(
             transmon_create(n),
             transmon_destroy(n),
         )
-        return 2 * np.pi * self.coupling / giga * op
+        return 2 * np.pi * op / giga
+
+    def operator(self, n: int) -> Operator:
+        """Time independent operator."""
+        if self.coupler is None:
+            return self.coupling * self._operator(n)
+
+        dim = [n, n, n]
+        op = expand(self.coupling * self._operator(n), dim, (0, 1))
+        op += expand(self.coupler.coupling[0] * self._operator(n=n), dim, (0, 2))
+        op += expand(self.coupler.coupling[1] * self._operator(n=n), dim, (1, 2))
+        op += expand(self.coupler.operator(n=n), dim, 2)
```

Comment:
With reference to the former comment, maybe it's just easier to keep three different terms (q1-q2, c-q1, c-q2), and the coupler operator could just be introduced by its presence as an ordinary `Qubit`.

Reply:
Given how the `Hamiltonian` is constructed in the current emulator I would suggest to keep the current implementation